person.rb

```diff
@@ -2,17 +2,30 @@
 require_relative 'decorator'
 require_relative 'capitalize_decorator'
 require_relative 'trimmer_decorator'
-
-# Class representing a person
-class Person
-  attr_accessor :name, :age
-  attr_reader :id
+require_relative 'classroom'
+require_relative 'rental'
+c
+# This Class represent a person
+class Person < Nameable
+  attr_accessor :name, :age, :rentals
+  attr_reader :id, :classroom
 
   def initialize(name, age = 'Unknown', parent_permission: true)
+    super()
     @id = rand(1..1000)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
+  end
+
+  def classroom=(classroom)
+    @classroom = classroom
+    classroom.add_student(self)
+  end
+
+  def add_rentals(book, date)
+    @rentals << Rental.new(date, book, self)
```

Comment:
- Hey, great using the `@rentals << Rental.new(date, book, self):` to append a new Rental object to the @rentals array to keep a collection of Rental objects for later use or processing, however, this would be also be achieved when you push into the @rentals array when you're adding the basic UI, in order not to repeat yourself, you should leave it as `Rental.new(date, book, self)`.